lib/ecto/migrator.ex

```diff
@@ -190,7 +190,19 @@ defmodule Ecto.Migrator do
     else
       {:ok, result} =
         repo.transaction(
-          fn -> send_and_receive(parent, ref, fun.()) end,
+          fn ->
+            if function_exported?(module, :after_begin, 0) do
+              module.after_begin()
```

Comment:
This will definitely execute in the context of the migration transaction, right?

Reply:
Yes.